src/drakvuf.cpp

```diff
@@ -161,6 +161,46 @@ int drakvuf_c::start_plugins(const bool* plugin_list, const plugins_options* opt
     return 1;
 }
 
+int drakvuf_c::stop_plugins(const bool* plugin_list)
+{
+    for (int i = 0; i < __DRAKVUF_PLUGIN_LIST_MAX; i++)
+    {
+        if (plugin_list[i])
+        {
+            int rc = plugins->stop(static_cast<drakvuf_plugin_t>(i));
+            if (rc < 0)
+                return rc;
+        }
+    }
+
+    return 1;
+}
+
+void drakvuf_c::wait_stopped(const bool* plugin_list)
```

Comment:
Yea, this function is very strange and needs to be simplified. I also don't see why the term `wait` here, there is no async processing of calls to plugins, it's all single threaded.

Reply:
@tklengyel, `wait` doesn't assume async operation. This is an active wait.

In case of async wait it should be called `await`.

The function's body is simple:
```c++
    GThread* timeout_thread = startup_timer(this, 20);
    drakvuf_loop(drakvuf, thunk, (void*)&is_plugins_stopped);
    cleanup_timer(this, timeout_thread);
```

Before that is a lambda for C-callback: https://bannalia.blogspot.com/2016/07/passing-capturing-c-lambda-functions-as.html.

Reply:
This as-is will not be merged into drakvuf main. As I said, this type of C++ is only OK from my PoV in plugins or helper libraries such as libusermode. You also didn't answer my question, why would you want to call this function? It implies plugins stop asynchronously and you want to wait for them to stop. Any call to a `stop` function will return only if the plugin is indeed stopped. So what does this wait do? What does it wait *for*?

Reply:
@tklengyel, I've refactored the code and simplified the C++ as You asked.

Also I have renamed the function too.

About Your main question why one would like to wait until plug-ins stop. If You examine some plugins (e.g. filedelete, memdump, procdump) You could see that it is not enough to remove traps. It could break VM on return path after injection. So we should wait until some job would be completed. Or better mark them to complete as soon as possible to not wait full file or memory dump (there are some work in progress). So we call `stop` and then wait (with polling) until plug-in notifies it have been successfully stopped. So this is the initial design.

Are there any other questions I could to clarify to support the PR?